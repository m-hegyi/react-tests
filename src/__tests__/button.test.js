import React from "react";
import renderer from "react-test-renderer";
import Button from "../components/button";

it("Button changes class when hovered", () => {
  const component = renderer.create(<Button />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  renderer.act(() => tree.props.onMouseEnter());
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  renderer.act(() =>Â tree.props.onMouseLeave());
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("Render the proper title", () => {
  const component = renderer.create(<Button title="hello world" />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


