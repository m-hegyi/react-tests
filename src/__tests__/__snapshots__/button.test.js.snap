// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button changes class when hovered 1`] = `
<button
  className="btn"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  
</button>
`;

exports[`Button changes class when hovered 2`] = `
<button
  className="btn btn-active"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  
</button>
`;

exports[`Button changes class when hovered 3`] = `
<button
  className="btn"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  
</button>
`;
